packages/utils/src/cronjob.ts

```diff
@@ -0,0 +1,104 @@
+import { JsonRpcRequestStruct } from '@metamask/utils';
+import {
+  array,
+  assign,
+  coerce,
+  create,
+  Infer,
+  object,
+  omit,
+  optional,
+  partial,
+  pick,
+  refine,
+  string,
+} from 'superstruct';
+import { parseExpression } from 'cron-parser';
+
+export const CronjobRpcRequestStruct = assign(
+  partial(pick(JsonRpcRequestStruct, ['id', 'jsonrpc'])),
+  omit(JsonRpcRequestStruct, ['id', 'jsonrpc']),
+);
+export type CronjobRpcRequest = Infer<typeof CronjobRpcRequestStruct>;
+
+export const CronExpressionStruct = refine(
+  coerce(
+    string(),
+    object({
+      minute: optional(string()),
+      hour: optional(string()),
+      dayOfMonth: optional(string()),
+      month: optional(string()),
+      dayOfWeek: optional(string()),
+    }),
+    (value) =>
+      `${value.minute ?? '*'} ${value.hour ?? '*'} ${value.dayOfMonth ?? '*'} ${
+        value.month ?? '*'
+      } ${value.dayOfWeek ?? '*'}`,
+  ),
+  'CronExpression',
+  (value) => {
+    try {
+      parseCronExpression(value);
+      return true;
+    } catch {
+      return false;
+    }
+  },
+);
+
+export type CronExpression = Infer<typeof CronExpressionStruct>;
+
+/**
+ * Parses a cron expression.
+ *
+ * @param expression - Expression to parse.
+ * @returns A CronExpression class instance.
+ */
+export function parseCronExpression(expression: string) {
+  return parseExpression(expression);
+}
+
+export const CronjobSpecificationStruct = object({
+  expression: CronExpressionStruct,
+  request: CronjobRpcRequestStruct,
+});
+export type CronjobSpecification = Infer<typeof CronjobSpecificationStruct>;
+
+/**
+ * Check if the given value is a {@link CronjobSpecification} object.
+ *
+ * @param value - The value to check.
+ * @returns Whether the value is a valid {@link CronjobSpecification} object.
+ */
+export function isCronjobSpecification(
+  value: unknown,
+): value is CronjobSpecification {
+  try {
+    create(value, CronjobSpecificationStruct);
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+export const CronjobSpecificationArrayStruct = array(
+  CronjobSpecificationStruct,
+);
+
+/**
+ * Check if the given value is an array of {@link CronjobSpecification} objects.
+ *
+ * @param value - The value to check.
+ * @returns Whether the value is a valid array of {@link CronjobSpecification} objects.
+ */
+export function isCronjobSpecificationArray(
+  value: unknown,
+): value is CronjobSpecification[] {
+  try {
+    create(value, CronjobSpecificationArrayStruct);
```

Comment:
I think you can just use `is` here.

Reply:
As we discussed and made conclusion, the only proper way to do this is by using `create`.

```
Note: You must use create(value, Struct) on the value to have the coercion take effect! Using simply assert() or is() will not use coercion.
```